Wikipedia/Code/StorageAndSyncingSettingsViewController.swift

```diff
@@ -0,0 +1,213 @@
+    }
+}
+
+@objc(WMFStorageAndSyncingSettingsViewController)
+class StorageAndSyncingSettingsViewController: UIViewController {
+    private let customViewCellReuseIdentifier = "CustomViewTableViewCell"
+    private var theme: Theme = Theme.standard
+    private var tableView: UITableView!
+    @objc public var dataStore: MWKDataStore?
+    private var indexPathsForCellsWithSwitches: [IndexPath] = []
+    
+    private var sections: [SettingsSection] {
+        let syncSavedArticlesAndLists = StorageAndSyncingSettingsItemType.syncSavedArticlesAndLists.settingsItem(isSwitchOnByDefault: isSyncEnabled)
+        let showSavedReadingList = StorageAndSyncingSettingsItemType.showSavedReadingList.settingsItem(isSwitchOnByDefault: dataStore?.readingListsController.isDefaultListEnabled)
+        let eraseSavedArticles = StorageAndSyncingSettingsItemType.eraseSavedArticles.settingsItem()
+        let syncWithTheServer = StorageAndSyncingSettingsItemType.syncWithTheServer.settingsItem()
+        
+        let syncSavedArticlesAndListsSection = SettingsSection(headerTitle: nil, footerText: WMFLocalizedString("settings-storage-and-syncing-enable-sync-footer-text", value: "Allow Wikimedia to save your saved articles and reading lists to your user preferences when you login to sync", comment: "Footer text of the settings option that enables saved articles and reading lists syncing"), items: [syncSavedArticlesAndLists])
+        
+        let showSavedReadingListSection = SettingsSection(headerTitle: nil, footerText: WMFLocalizedString("settings-storage-and-syncing-show-default-reading-list-footer-text", value: "Show the Saved (eg. default) reading list as a separate list in your Reading lists view. This list appears on Android devices", comment: "Footer text of the settings option that enables showing the default reading list"), items: [showSavedReadingList])
+        
+        let eraseSavedArticlesSection = SettingsSection(headerTitle: nil, footerText: nil, items: [eraseSavedArticles])
+        let syncWithTheServerSection = SettingsSection(headerTitle: nil, footerText: WMFLocalizedString("settings-storage-and-syncing-server-sync-footer-text", value: "Request a sync from the server for an update to your synced articles and reading lists", comment: "Footer text of the settings button that initiates saved articles and reading lists server sync"), items: [syncWithTheServer])
+        
+        return [syncSavedArticlesAndListsSection, showSavedReadingListSection, eraseSavedArticlesSection, syncWithTheServerSection]
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = WMFLocalizedString("settings-storage-and-syncing-title", value: "Saved articles storage and syncing", comment: "Title of the saved articles storage and syncing settings screen")
```

Comment:
Something weird is happing when this text appears and after rotate - it jumps a bit on appearance and is vertically offset a bit after rotate.